FirebaseSessions/Tests/Unit/FirebaseSessionsTests.swift

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+#if SWIFT_PACKAGE
+  import FirebaseSessionsObjC
+#endif // SWIFT_PACKAGE
+
+@testable import FirebaseSessions
+
+///
+/// This is the parent class for tests that test against the Sessions internal
+/// API or test Sessions `init` functionality.
+///
+/// Test cases should only go in subclasses because all tests in the parent
+/// class will be run by subclasses.
+///
+class FirebaseSessionsTests: XCTestCase {
```

Comment:
Why is this extended out of XCTestCase while most of the job done by this class is Utility driven? I would go to the extend of calling it FirebaseSessionsMock.

Reply:
Great question - I wanted to be able to reference the instance variables, so I made this a base class. It subclasses XCTestCase so that other classes can be subclasses of it.

I think maybe renaming it to FirebaseSessionsTestsBase might help convey what I meant.

Or I might go with encapsulation over subclassing in this case as you suggested